### PR TITLE
Openapi operationid

### LINKED ISSUE #6: Custom id generation strategy when generating openapi docs

### PR DESCRIPTION
**Pull request by ghostbuster91**
This closes #6

### CODE REVIEW

File: docs/openapi-docs/src/test/scala/tapir/docs/openapi/VerifyYamlTest.scala [hunk 1]
```diff
@@ -30,15 +30,34 @@ class VerifyYamlTest extends FunSuite with Matchers {
...
     actualYamlNoIndent shouldBe expectedYaml
   }
 
-  case class F1(data: List[F1])
+  test("should use custom operationId generator") {
+    def customOperationIdGenerator(pc: Vector[String], m: Method) = pc.map(_.toUpperCase).mkString("", "+", "-") + m.m.toUpperCase
+    val options = OpenApiDocsOptions.Default.copy(customOperationIdGenerator)
+    val expectedYaml = loadYaml("expected_custom_operation_id.yml")
+
+    val actualYaml = RichOpenAPI(
```
Comment (adamw) [inline docs/openapi-docs/src/test/scala/tapir/docs/openapi/VerifyYamlTest.scala]: why are all implicits explicit?

File: docs/openapi-docs/src/main/scala/tapir/docs/openapi/EndpointToOpenApiPaths.scala [hunk 2]
```diff
@@ -5,7 +5,7 @@ import tapir.openapi.OpenAPI.ReferenceOr
 import tapir.openapi.{MediaType => OMediaType, _}
 import tapir.{EndpointInput, MediaType => SMediaType, _}
 
-private[openapi] class EndpointToOpenApiPaths(objectSchemas: ObjectSchemas) {
+private[openapi] class EndpointToOpenApiPaths(objectSchemas: ObjectSchemas, options: OpenApiDocsOptions) {
```
Comment (adamw) [inline docs/openapi-docs/src/main/scala/tapir/docs/openapi/EndpointToOpenApiPaths.scala]: [minor] for consistency, the name should be `OpenAPIDocsOptions` :)

File: docs/openapi-docs/src/test/scala/tapir/docs/openapi/VerifyYamlTest.scala (hunk 1 above)
Comment (ghostbuster91) [reply docs/openapi-docs/src/test/scala/tapir/docs/openapi/VerifyYamlTest.scala]: Was checking something using `desugar` feature and forget to roll it back. Thanks for the caution.

**Comment by adamw** [comment]
Thanks! :)

**Pull request merged**